### PR TITLE
Cleanup method names

### PR DESCRIPTION
**Pull request by sduffy77**
GetAllRoles method name doesn't match it's function
In the RoleExtensions class there is a GetAllRoles method which returns a role by Id. This is misleading based on the method name.
UpdateReviewComponents method name doesn't match it's function
In the ReviewExtensions class the UpdateReviewComponents method which retrieves a list of changed Review Components. This is misleading based on the method name.

The "Get..." retrieval naming convention is very Java-esk and Rob would like to see these methods renamed to "Find..." and obsolete the "Get..." methods. Make each "Get..." method call the corresponding "Find..." method.

### CODE REVIEW

File: src/Configuration/ConfigurationExtensions.cs [hunk 1]
```diff
@@ -448,6 +608,18 @@ public static List<PluginParameterModel> GetExportPluginParameters(this InspireC
         /// <param name="exportType">Contains the <see cref="ExportType"/> type that will be retrieved, Standard is default.</param>
         /// <returns>Returns a list of <see cref="ExportSelectionModel"/> objects from the application.</returns>
         public static ExportSelectionModel GetExportComponentType(this InspireClient client, long componentTypeId, ExportType exportType = ExportType.Standard)
```
Comment (timaxapa) [inline src/Configuration/ConfigurationExtensions.cs:610]: Did we forget to add `[Obsolete]` attribute here?

File: src/Review/ReviewExtensions.cs [hunk 2]
```diff
@@ -105,7 +191,18 @@ public static List<ReviewAssignmentModel> GetUserReviewAssignments(this InspireC
...
         /// <param name="client"><see cref="InspireClient"/> used to communication with the API endpoint.</param>
         /// <returns>Returns a list of <see cref="MinimalUserModel"/> objects if found.</returns>
+        [Obsolete("This method is obsolete. Please use RetrieveReviewCoordinators() going forward. This method will be removed in a future release.")]
         public static List<MinimalUserModel> GetReviewCoordinators(this InspireClient client)
+        {
+            return RetrieveReviewCoordinators(client);
+        }
+
+        /// <summary>
+        /// This method retrieves the users that that are coordinators.
```
Comment (timaxapa) [inline src/Review/ReviewExtensions.cs:201]: Duplicated `that` word

File: src/Review/ReviewExtensions.cs [hunk 3]
```diff
@@ -129,7 +226,19 @@ public static ReviewModel UpdateReview(this InspireClient client, ReviewModel mo
...
+        [Obsolete("This method is obsolete. Please use FindChangedReviewComponents")]
         public static List<ReviewerComponentModel> UpdateReviewComponents(this InspireClient client, ReviewChangedComponentQueryModel model)
+        {
+            return FindChangedReviewComponents(client, model);
+        }
+
+        /// <summary>
+        /// This method is used to return review components within a review that have been updated after a specified date.
+        /// </summary>
+        /// <param name="client">The client.</param>
```
Comment (timaxapa) [inline src/Review/ReviewExtensions.cs:238]: In all other methods we use the following description for `client` parameter. Should we use it here too?
```
/// <param name="client"><see cref="InspireClient"/> used to communication with the API endpoint.</param>
```

File: src/Security/RoleExtensions.cs [hunk 4]
```diff
@@ -19,31 +19,57 @@ public static class RoleExtensions
...
         {
-            var request = client.CreateRequest($"/Roles?roleName={roleName}&orderBy={orderBy}&direction={direction}");
+            var request = client.CreateRequest($"/Roles/{roleName}/{orderBy}/{direction}");
             return client.RequestContent<List<RoleModel>>(request);
         }
 
+        /// <summary>
+        /// Gets the specified <see cref="RoleModel"/>.
+        /// </summary>
+        /// <param name="client">The client.</param>
```
Comment (timaxapa) [inline src/Security/RoleExtensions.cs:52]: In all other methods we use the following description for `client` parameter. Should we use it here too?
```
/// <param name="client"><see cref="InspireClient"/> used to communication with the API endpoint.</param>
```

File: src/Security/RoleExtensions.cs [hunk 5]
```diff
@@ -19,31 +19,57 @@ public static class RoleExtensions
...
+        [Obsolete("This method is obsolete. Please use FindRole() going forward. This method will be removed in a future release.")]
+        public static RoleModel GetRole(this InspireClient client, long roleId)
+        {
+            return FindRole(client, roleId);
+        }
+
+        /// <summary>
+        /// Gets the specified <see cref="RoleModel"/>.
+        /// </summary>
+        /// <param name="client">The client.</param>
```
Comment (timaxapa) [inline src/Security/RoleExtensions.cs:64]: In all other methods we use the following description for `client` parameter. Should we use it here too?
```
/// <param name="client"><see cref="InspireClient"/> used to communication with the API endpoint.</param>
```

**Review by timaxapa — CHANGES_REQUESTED** [review]

File: src/Configuration/ConfigurationExtensions.cs (hunk 1 above)
Comment (sduffy77) [reply src/Configuration/ConfigurationExtensions.cs:610]: Indeed we did, good catch. Added

File: src/Review/ReviewExtensions.cs (hunk 2 above)
Comment (sduffy77) [reply src/Review/ReviewExtensions.cs:201]: Fixed typos

File: src/Review/ReviewExtensions.cs (hunk 3 above)
Comment (sduffy77) [reply src/Review/ReviewExtensions.cs:238]: Replaced description

File: src/Security/RoleExtensions.cs (hunk 4 above)
Comment (sduffy77) [reply src/Security/RoleExtensions.cs:52]: Replaced description

File: src/Security/RoleExtensions.cs (hunk 5 above)
Comment (sduffy77) [reply src/Security/RoleExtensions.cs:64]: Replaced description.

**Comment by sduffy77** [comment]
Did a search for "The client" and replaced with consistent param description

**Review by timaxapa — APPROVED** [review]

**Pull request merged**